src/filing/institutions/container.jsx

```diff
@@ -34,8 +43,28 @@ export class InstitutionContainer extends Component {
     }
   }
 
+  updateUrlIfNeeded(){
```

Comment:
I think that instead of managing the url ad hoc here, that this should be incorporated into the route for the institution page directly. Namely, I think an `/:institution` argument should be added [in the index file](https://github.com/cfpb/hmda-frontend/blob/master/src/filing/index.js#L62) which then can be passed down and used by the institution container to control routing. The info button just becomes a `Link` to `/filing/<year>/institutions/<institution/` ( I think we can drop `/details` since it's kinda self-evident). And the back button from the details page is just `Link` back to `/filing/<year>/institutions/`

Reply:
Much cleaner and more sensible than the fiasco I pieced together.  🤦🏽  
 
Thank you!  🚀